src/test/java/io/redis/examples/async/StringExample.java

```diff
@@ -0,0 +1,127 @@
+// EXAMPLE: set_tutorial
+package io.redis.examples.async;
+
+import io.lettuce.core.*;
+import io.lettuce.core.api.async.RedisAsyncCommands;
+import io.lettuce.core.api.StatefulRedisConnection;
+
+// REMOVE_START
+import org.junit.jupiter.api.Test;
+// REMOVE_END
+
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+
+// REMOVE_START
+import static org.assertj.core.api.Assertions.assertThat;
+// REMOVE_END
+
+public class StringExample {
+
+    // REMOVE_START
+    @Test
+    // REMOVE_END
+    public void run() {
+        RedisClient redisClient = RedisClient.create("redis://localhost:6379");
+
+        try (StatefulRedisConnection<String, String> connection = redisClient.connect()) {
+            RedisAsyncCommands<String, String> asyncCommands = connection.async();
+
+            // STEP_START set_get
+            CompletableFuture<Void> setAndGet = asyncCommands.set("bike:1", "Deimos").thenCompose(v -> {
+                System.out.println(v); // OK
+                // REMOVE_START
+                assertThat(v).isEqualTo("OK");
+                // REMOVE_END
+                return asyncCommands.get("bike:1");
+            })
+                    // REMOVE_START
+                    .thenApply(res -> {
+                        assertThat(res).isEqualTo("Deimos");
+                        return res;
+                    })
+                    // REMOVE_END
+                    .thenAccept(System.out::println) // Deimos
+                    .toCompletableFuture();
+            // STEP_END
```

Comment:
```suggestion
// STEP_START set_get
            // Deimos
            Mono<String> setAndGet = reactiveCommands.set("bike:1", "Deimos")
                    .doOnNext(res-> System.out.println(res)) // OK
                    .flatMap(v -> reactiveCommands.get("bike:1"))
                    .doOnNext(res-> System.out.println(res));  // Deimos

            // REMOVE_START
            String[] output = SysOutCaptor.CaptureSystemOut(() ->
                    StepVerifier.create(setAndGet)
                            .assertNext(v -> assertThat(v).isEqualTo("Deimos"))
                            .expectComplete()
                            .verify()
            );
            assertThat(output).containsSequence("OK", "Deimos");
            // REMOVE_END
            // STEP_END
```
I suggest extracting checks for example code as separate step for readbility.